File: 3435.java
/* @author: */
import java.util.Scanner;
import java.util.Arrays;
public class Main{

 public static void main(String args[])
{
 Scanner sc=new Scanner(System.in);

 int n,i,j,k1,k2,p[][]=new int[101][101];
 n=sc.nextInt();

 int len=n*n;
 for(i=0;i< len;i++)
   for(j=0;j< len;j++)
    p[i][j]=sc.nextInt();
 boolean bb[]=new boolean[101],is=true;
  //memset(bb,0,sizeof(bb));
 for(i=0;i< len;i++)
 {
    Arrays.fill(bb,false);
    for(j=0;j< len;j++)
    {
         if(p[i][j]==0)continue;
      if(bb[p[i][j]])break;
      bb[p[i][j]]=true;
    }
    if(j< len)break;
   }
  if(i< len)is=false;
  for(j=0;j< len;j++)
  {
    Arrays.fill(bb,false);
    for(i=0;i< len;i++)
    {
         if(p[i][j]==0)continue;
      if(bb[p[i][j]])break;
      bb[p[i][j]]=true;
    }
    if(i< len)break;
   }
   if(j< len)is=false;
   for(i=0;i< len;i+=n)
    {
    for(j=0;j< len;j+=n)
    {
       Arrays.fill(bb,false);
       for(k1=i;k1< i+n;k1++)
       {
        for(k2=j;k2< j+n;k2++)
        {
             if(p[k1][k2]==0)continue;
          if(bb[p[k1][k2]])break;
          bb[p[k1][k2]]=true;
        }
        if(k2!=j+n)break;
        }
       if(k1!=i+n)is=false;
       if(is==false)break;
    }
    if(is==false)break;
     }
     if(is) System.out.println("CORRECT");
     else System.out.println("INCORRECT");
   }
}

File: 3435.cc
#include<cstdio>
#include<cstring>
#define MAXN 105
int N, arr[MAXN][MAXN];
int x_used[MAXN][11], y_used[MAXN][11], n_used[MAXN][11];

int main() {
    while(scanf("%d", &N) != EOF) {
        memset(x_used, 0, sizeof(x_used));
        memset(y_used, 0, sizeof(y_used));
        memset(n_used, 0, sizeof(n_used));
        bool flag = true;
        for(int i = 0; i < N * N; ++ i) {
            for(int j = 0; j < N * N; ++ j) {
                scanf("%d", &arr[i][j]);
                if(arr[i][j] > 0 && flag) {
                    if(x_used[i][arr[i][j]] == 1) {
                        flag = false;
                    } else {
                        x_used[i][arr[i][j]] = 1;
                    }
                    if(y_used[j][arr[i][j]] == 1) {
                        flag = false;
                    } else {
                        y_used[j][arr[i][j]] = 1;
                    }
                    if(n_used[i / N * N + j / N + 1][arr[i][j]] == 1) {
                        flag = false;
                    } else {
                        n_used[i / N * N + j / N + 1][arr[i][j]] = 1;
                    }
                }
            }
        }
        if(flag) {
            printf("CORRECT\n");
        } else {
            printf("INCORRECT\n");
        }
    }
    return 0;
}



